tests/reference/asr-arrays_01-a617b64.stderr

```diff
@@ -2,7 +2,7 @@ semantic error: Type mismatch in procedure call; the types must be compatible
   --> tests/errors/arrays_01.py:15:9
    |
 15 |         [i8(214), i8(157), i8(3), i8(146)])
-   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type mismatch (passed argument type is list[i8] but required type is i8[4])
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type mismatch (passed argument type is list[i8] but required type is i8[0,4])
    |
  9 |     a : i8[4] = empty(4, dtype=int8)
-   |         ^^^^^ type mismatch (passed argument type is list[i8] but required type is i8[4])
+   |         ^^^^^ type mismatch (passed argument type is list[i8] but required type is i8[0,4])
```

Comment:
This should be
```suggestion
   |         ^^^^^ type mismatch (passed argument type is list[i8] but required type is i8[4])
```
right?

Reply:
We previously only supported encoding a single dimension to string. Now we support encoding multi-dimensions to string. We also encode the start index or lower bound which is more descriptive.

Reply:
If this is okay with @certik, Let's merge this PR.